psi4/run_psi4.py

```diff
@@ -74,6 +74,8 @@
                     help="Runs a JSON input file. !Warning! experimental option.")
 parser.add_argument("-t", "--test", nargs='?', const='smoke', default=None,
                     help="Runs pytest tests. If `pytest-xdist` installed, parallel with `--nthread`.")
+parser.add_argument("--mdi", default=None,
```

Comment:
should this mirror `--json` as much as possible?

Reply:
I'd rewrite to say this creates and launches a MDI listener (as opposed to just "configuring" one). Also, the `scf_method` argument should be passed here, correct?

Reply:
Actually, upon re-reading this, I have no idea how the `--mdi` command line option works. The test only covers setting/getting values out of/into psi4 using the python interface. Do you have an example snippet of what it looks like in practice?

Reply:
The `mdi('hf/cc-pvdz')` is a little scary, as we try to keep the top-level commands constant (energy/gradient/optimize/hessian/frequency) even as their capabilities increase. A `energy('hf/cc-pvdz', mdi=True)` would be better from the perspective of normal Psi4 behavior. Can we revisit this after the distributed driver merge or before you start advertising the mdi/psi interface?

Reply:
I do agree with @loriab on this. I'd rather have the `energy` function be `mdi`-aware than add a top-level command.  I think once we add `psi4.mdi()` it'll be hard to get rid of it...

Actually, the only thing the `psi4.mdi()` function does that cannot be done on init is setting the method string. I'm not sure whether the MDI standard provides a way to set this using a command? Or does the standard require a `mdi` function?